pkg/wifi/iwl.go

```diff
@@ -195,20 +194,23 @@ func (w *IWLWorker) Connect(stdout, stderr io.Writer, a ...string) error {
 
 	}()
 
-	//	select {
-	//	case err := <-c:
-	//		return err
-	//	case <-ctx.Done():
-	//		return fmt.Errorf("wpa_supplicant error")
-	//	}
-
 	// dhclient might never return on incorrect passwords or identity
-	cmd := exec.CommandContext(ctx, "dhclient", "-ipv4=true", "-ipv6=false", "-v", w.Interface)
-	cmd.Stdout, cmd.Stderr = stdout, stderr
-	if err := cmd.Run(); err != nil {
+	go func() {
+		cmd := exec.CommandContext(ctx, "dhclient", "-ipv4=true", "-ipv6=false", "-v", w.Interface)
+		cmd.Stdout, cmd.Stderr = stdout, stderr
+		if err := cmd.Run(); err != nil {
+			c <- err
+		} else {
+			c <- nil
+		}
+	}()
+
+	select {
+	case err := <-c:
 		return err
+	case <-ctx.Done():
+		return fmt.Errorf("wpa_supplicant error")
```

Comment:
The wpa_supplicant doesn't timeout out anymore, but dhclient does. Perhaps this error should be "dhcp timeout".